src/main/java/seedu/address/logic/commands/RequiredCommand.java

```diff
@@ -105,11 +129,13 @@ public boolean doesModuleAlreadyExist(Module module, ObservableList<Module> modu
      * @param failMessage Fail message for particular category of Modules.
      * @param successMessage Success message for particular category of Modules.
      */
-    public void compareModules(ObservableList<Module> modules, String failMessage, String successMessage) {
+    public void compareModules(ObservableList<Module> modules, Map<String, String> preclusionMap,
+                               String failMessage, String successMessage) {
```

Comment:
Just curious why did you put `preclusionMap` as a parameter? Can it be stored as a field and accessed through the class instead since there is only 1 preclusionMap. I'm not too sure I'm just asking

Reply:
Good point, I'm unsure if I should store it as a class variable because idk if the RequiredCommand object should deal with the preclusionMap storage. But it does make the code a bit cleaner if I store it as well HAHAHA

Reply:
Actually we don't need to store it separately since we already store `storage`, which contains the map right HAHA

And just to add on, notice that `execute` and `compareModules` are now aware of `preclusionMap`, even though it is only used by `isModuleAPreclusion`, a much lower-level method. The parameter is being passed through 2 methods despite them never using it. This anti-pattern is also called prop drilling/threading in JS btw HAHA

Maybe we can just do `Map<String, String> preclusionMap = storage.getPreclusionMap()` within the `isModuleAPreclusion` fxn instead of in `execute`?

Reply:
Good point....I totally forgot Im already storing it in storage HAHAH changed accordingly!